iam_role.tf

```diff
@@ -0,0 +1,68 @@
+provider "aws" {
+  region = var.aws_region
+}
+
+resource "aws_iam_openid_connect_provider" "github" {
+  url = "https://token.actions.githubusercontent.com"
+
+  client_id_list = ["sts.amazonaws.com"]
+  thumbprint_list = ["6938fd4d98e9a3b8a5b1baba04a40f10a11123d6"]
+}
+
+resource "aws_iam_role" "github_actions_role" {
+  name = "GithubActionsRole"
+
+  assume_role_policy = jsonencode({
```

Comment:
Also, [aws_iam_policy_document](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/iam_policy_document) can be considered for better linter support